### ISSUE TITLE
Android - Leaked db cursor

### ISSUE THREAD

**Issue by mortenholmgaard** [Adobe-Marketing-Cloud/acp-sdks#280]
I get a few LeakCanary warning about leaks on every run - please have a look at it.

```
Finalizing a Cursor that has not been deactivated or closed. database = /data/user/0/myBundleIdentifer/cache/ADBMobileDataCache.sqlite, table = HITS, query = SELECT ID FROM HITS
    android.database.sqlite.DatabaseObjectNotClosedException: Application did not close the cursor or database object that was opened here
        at android.database.sqlite.SQLiteCursor.<init>(SQLiteCursor.java:98)
        at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:50)
        at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1628)
        at android.database.sqlite.SQLiteDatabase.queryWithFactory(SQLiteDatabase.java:1475)
        at android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1346)
        at android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1552)
        at com.adobe.marketing.mobile.AndroidDatabase.query(AndroidDatabase.java:116)
        at com.adobe.marketing.mobile.AbstractHitsDatabase.getSize(AbstractHitsDatabase.java:198)
        at com.adobe.marketing.mobile.AbstractHitsDatabase.getSize(AbstractHitsDatabase.java:179)
        at com.adobe.marketing.mobile.HitQueue.updateAllHits(HitQueue.java:224)
        at com.adobe.marketing.mobile.AnalyticsHitsDatabase.kick(AnalyticsHitsDatabase.java:312)
        at com.adobe.marketing.mobile.AnalyticsExtension.updatePrivacyStatus(AnalyticsExtension.java:369)
        at com.adobe.marketing.mobile.AnalyticsExtension.handleConfigurationUpdateEvent(AnalyticsExtension.java:1137)
        at com.adobe.marketing.mobile.AnalyticsListenerConfigurationResponseContent$1.run(AnalyticsListenerConfigurationResponseContent.java:45)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:762)
2019-11-18 08:13:49.898 16691-16700/myBundleIdentifer E/StrictMode: Finalizing a Cursor that has not been deactivated or closed. database = /data/user/0/myBundleIdentifer/cache/ADBMobileIdentity.sqlite, table = REQUESTS, query = SELECT ID FROM REQUESTS
    android.database.sqlite.DatabaseObjectNotClosedException: Application did not close the cursor or database object that was opened here
        at android.database.sqlite.SQLiteCursor.<init>(SQLiteCursor.java:98)
        at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:50)
        at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1628)
        at android.database.sqlite.SQLiteDatabase.queryWithFactory(SQLiteDatabase.java:1475)
        at android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1346)
        at android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1552)
        at com.adobe.marketing.mobile.AndroidDatabase.query(AndroidDatabase.java:116)
        at com.adobe.marketing.mobile.AbstractHitsDatabase.getSize(AbstractHitsDatabase.java:198)
        at com.adobe.marketing.mobile.AbstractHitsDatabase.getSize(AbstractHitsDatabase.java:179)
        at com.adobe.marketing.mobile.HitQueue.updateAllHits(HitQueue.java:224)
        at com.adobe.marketing.mobile.IdentityHitsDatabase.resetEventNumberAndPairIdForExistingRequests(IdentityHitsDatabase.java:275)
        at com.adobe.marketing.mobile.IdentityHitsDatabase.<init>(IdentityHitsDatabase.java:90)
        at com.adobe.marketing.mobile.IdentityHitsDatabase.<init>(IdentityHitsDatabase.java:55)
        at com.adobe.marketing.mobile.IdentityExtension.initializeDatabaseWithCurrentPrivacyStatus(IdentityExtension.java:228)
        at com.adobe.marketing.mobile.IdentityExtension.handleSyncIdentifiers(IdentityExtension.java:607)
        at com.adobe.marketing.mobile.IdentityExtension.processEvent(IdentityExtension.java:484)
        at com.adobe.marketing.mobile.IdentityExtension.processEventQueue(IdentityExtension.java:168)
        at com.adobe.marketing.mobile.ListenerHubSharedStateIdentity$1.run(ListenerHubSharedStateIdentity.java:51)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:762)
```

```
 A resource was acquired at attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks.
    java.lang.Throwable: Explicit termination method 'close' not called
        at dalvik.system.CloseGuard.open(CloseGuard.java:180)
        at android.database.CursorWindow.<init>(CursorWindow.java:111)
        at android.database.AbstractWindowedCursor.clearOrCreateWindow(AbstractWindowedCursor.java:198)
        at android.database.sqlite.SQLiteCursor.clearOrCreateWindow(SQLiteCursor.java:300)
        at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:138)
        at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:132)
        at com.adobe.marketing.mobile.AndroidCursor.getCount(AndroidCursor.java:49)
        at com.adobe.marketing.mobile.AbstractHitsDatabase.getSize(AbstractHitsDatabase.java:205)
        at com.adobe.marketing.mobile.AbstractHitsDatabase.getSize(AbstractHitsDatabase.java:179)
        at com.adobe.marketing.mobile.HitQueue.updateAllHits(HitQueue.java:224)
        at com.adobe.marketing.mobile.AnalyticsHitsDatabase.kick(AnalyticsHitsDatabase.java:312)
        at com.adobe.marketing.mobile.AnalyticsExtension.updatePrivacyStatus(AnalyticsExtension.java:369)
        at com.adobe.marketing.mobile.AnalyticsExtension.handleConfigurationUpdateEvent(AnalyticsExtension.java:1137)
        at com.adobe.marketing.mobile.AnalyticsListenerConfigurationResponseContent$1.run(AnalyticsListenerConfigurationResponseContent.java:45)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:762)
```

**Comment by kc2cheng** [comment]
Hey Morten, thanks for your report.  Our engineering team will take a look.

**Comment by nporter-adbe** [comment]
Hi @mortenholmgaard we've identified the potential cause for this cursor leak and have submitted a fix which we hope to release soon.

We'll update this issue when the fix is released.

**Comment by nporter-adbe** [comment]
@mortenholmgaard This has been fixed in Core 1.5.0.
https://aep-sdks.gitbook.io/docs/release-notes#android-core-1-5-0
